src/fontman/raster_font.cpp

```diff
@@ -0,0 +1,490 @@
+                if(font)
+                {
+                    TtfFont::TheGlyphInfo glyph = font->getGlyphInfo(&cx, m_letterWidth);
+                    uint32_t glyph_width = glyph.width > 0 ? uint32_t(glyph.advance >> 6) : (m_letterWidth >> 2);
+                    auto lw = m_letterWidth + m_interLetterSpace;
+                    // Raster fonts are monospace fonts. TTF glyphs shoudn't break mono-width until they are wider than a cell
+                    widthSumm += glyph_width > lw ? glyph_width : lw;
+                }
+                else
+#endif
+                {
+                    widthSumm += m_letterWidth + m_interLetterSpace;
+                }
+            }
+
+            if(widthSumm > widthSummMax)
+                widthSummMax = widthSumm;
+
+            break;
+        }
+
+        }//Switch
+
+        if((max_line_lenght > 0) && (x >= max_line_lenght)) //If lenght more than allowed
+        {
+            maxWidth = x;
+            if(t != &modText)
+            {
+                modText = text;
+                t = &modText;
+            }
+
+            if(lastspace > 0)
+            {
+                modText[lastspace] = '\n';
+                i = lastspace - 1;
+                lastspace = 0;
+            }
+            else
+            {
+                modText.insert(i, 1, '\n');
+                x = 0;
+                count++;
+            }
+        }
+        i += static_cast<size_t>(trailingBytesForUTF8[uch]);
+    }
+
+    if(count == 1)
+        maxWidth = static_cast<uint32_t>(text.length());
+
+    /****************Word wrap*end*****************/
+    return PGE_Size(static_cast<int32_t>(widthSummMax), static_cast<int32_t>(m_newlineOffset * count));
+}
+
+void RasterFont::printText(const std::string &text,
+                           int32_t x, int32_t y,
+                           float Red, float Green, float Blue, float Alpha,
+                           uint32_t)
+{
+    if(m_charMap.empty() || text.empty())
+        return;
+
+    uint32_t offsetX = 0;
+    uint32_t offsetY = 0;
+    uint32_t w = m_letterWidth;
+    uint32_t h = m_letterHeight;
+#ifdef THEXTECH_ENABLE_TTF_SUPPORT
+    //! FIXME: Add the config option for TTF fonts here
+    bool    doublePixel = false; //ConfigManager::setup_fonts.double_pixled;
+#endif
+
+    const char *strIt  = text.c_str();
+    const char *strEnd = strIt + text.size();
+    for(; strIt < strEnd; strIt++)
+    {
+        const char &cx = *strIt;
+        UTF8 ucx = static_cast<unsigned char>(cx);
+
+        switch(cx)
+        {
+        case '\n':
+            offsetX = 0;
+            offsetY += m_newlineOffset;
+            continue;
+
+        case '\t':
+            //Fake tabulation
+            offsetX += offsetX + offsetX % w;
+            continue;
+
+        case ' ':
+            offsetX += m_spaceWidth + m_interLetterSpace / 2;
+            continue;
+        }
+
+        RasChar rch = m_charMap[get_utf8_char(strIt)];
```

Comment:
I think this mallocs an entire RasChar for every new failed char. Is that right? Maybe better to do a find call instead.

Reply:
Good advice, I'll try that.

Reply:
Done